src/System.Management.Automation/engine/ParameterBinderBase.cs

```diff
@@ -1244,8 +1244,9 @@ internal InternalCommand Command
                         // However, we don't allow Hashtable-to-Object conversion (PSObject and IDictionary) because
                         // those can lead to property setters that probably aren't expected. This is enforced by
                         // setting 'Context.LanguageModeTransitionInParameterBinding' to true before the conversion.
+                        var currentLanguageMode = Context.LanguageMode;
                         bool changeLanguageModeForTrustedCommand =
-                            Context.LanguageMode == PSLanguageMode.ConstrainedLanguage &&
+                            currentLanguageMode == PSLanguageMode.ConstrainedLanguage &&
                             this.Command.CommandInfo.DefiningLanguageMode == PSLanguageMode.FullLanguage;
```

Comment:
Shall we check on `currentLanguageMode == PSLanguageMode.ConstrainedLanguageAudit` too? `LanguageModeTransitionInParameterBinding` should be set to `true` if we are mimicking `ConstrainedLanguage` mode.